core/domain/activity_domain.py

```diff
@@ -28,17 +28,38 @@ class ActivityReference(object):
     """
 
     def __init__(self, activity_type, activity_id):
+        """Constructs an ActivityReference instance.
+
+        Args:
+          activity_type: str.
+          activity_id:   str.
+        """
         self.type = activity_type
         self.id = activity_id
 
     def get_hash(self):
+        """Gets instance hash.
```

Comment:
This doesn't help me -- reading this, I don't know what the "instance hash" is and what it's meant to be used for. Please look around the code (use grep) and try and figure out where this is used and what it's used for, then write the docstring so that a new developer can understand what the context is. Thanks!